components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -686,11 +686,14 @@ public UserRecoveryData introspectUserSelfRegistration(boolean skipExpiredCodeVa
                                                            String verifiedChannelClaim, Map<String, String> properties)
             throws IdentityRecoveryException {
 
-        return validateSelfRegistrationCode(code, verifiedChannelType, verifiedChannelClaim, properties, skipExpiredCodeValidation);
+        return validateSelfRegistrationCode(code, verifiedChannelType, verifiedChannelClaim,
+                properties, skipExpiredCodeValidation, true);
     }
 
     private UserRecoveryData validateSelfRegistrationCode(String code, String verifiedChannelType,
-                                                          String verifiedChannelClaim, Map<String, String> properties, boolean skipExpiredCodeValidation)
+                                                          String verifiedChannelClaim, Map<String, String> properties,
+                                                          boolean skipExpiredCodeValidation,
+                                                          boolean skipAccountConfirmation)
```

Comment:
can 't we use the recoveryScenario for this purpose instead of introducing a new variable.

Reply:
For introspection & validation, we are getting the same Recovery step: "RecoverySteps.CONFIRM_LITE_SIGN_UP" https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/RecoverySteps.java#L32. Thus it is hard to differentiate them